### ISSUE TITLE
Interpret \comment and \endcomment as commenting out

### ISSUE THREAD

**Issue by prowlett** [coursebuilder-ncl/makecourse#87]
I have a bit of LaTeX code in my exam paper to hide/show answers which works like this. It doesn't work. 

- if `THIS LINE TO HIDE ANSWERS` is used, all the answer text appears, i.e. it is not hidden.
- if `THIS LINE TO SHOW ANSWERS` is used, the answer text appears between `[Answer: ` and `]` as expected. 

The hidden version of this uses `\comment` and `\endcomment` from the `verbatim` package to lose a block of code. Is there any chance you could interpret `\comment` and `\endcomment` as skipping what comes between them? Or is there something `makecourse`-compliant that I could be doing here?

```
\documentclass{article}

\usepackage{verbatim}
\newenvironment{answer}{\comment}{\endcomment} % THIS LINE TO HIDE ANSWERS
%\newenvironment{answer}{[Answer: }{]} % THIS LINE TO SHOW ANSWERS

\begin{document}
    \begin{enumerate}
        \item Question \begin{answer}Answer\end{answer}
    \end{enumerate}
\end{document}
```

**Comment by georgestagg** [comment]
Looks like the `\comment` macro might be defined in the wrong place. Does it work if you add `\usepackage{comment}`?

**Comment by georgestagg** [comment]
Okay, thanks for testing that.

I looked into this in further detail this morning. The primary cause seems to be LaTeX and plasTeX implementing the `verbatim` package differently. When you call `\verbatim` (or `\comment` in this case) the token stream is searched for the corresponding `\endverbatim` and everything between the two is output as-is. Importantly, during this search macro expansion is turned off (because when you output verbatim content you don't want to expand anything, even if it begins with `\`).

So, after you invoke `\begin{answer}` the code was looking for a `\endcomment` that never comes. It only sees `\end{answer}`, but doesn't expand it and so misses the `\endcomment` hiding inside. That's why all the content after the "Question" was lost.

In the latest commit I've tweaked the implementation of `verbatim` to add an extra search for `\end{...}`. If it finds it, it expands and looks inside for the real `\end...`. Finally, if it finds the real thing, the `\end{...}` is expanded, otherwise it prints it verbatim and continues on as before.

As far as I can tell, the following example now works for me. Hopefully it's the same case for your real exam.
```
 \documentclass{article}
 
 \usepackage{verbatim}
 \newenvironment{answer}{\comment}{\endcomment} % THIS LINE TO HIDE ANSWERS
 %\newenvironment{answer}{[Answer: }{]} % THIS LINE TO SHOW ANSWERS
 
 \begin{document}
     Content before enumerate
     \begin{enumerate}
         \item Question \begin{answer}Answer\end{answer}
     \end{enumerate}
     Content after enumerate
 \end{document}
```
